### ISSUE TITLE
[BUG] Azure blob sdk v12 preview does not seem to include Etags in Blob Properties

### ISSUE THREAD

**Issue by raj2103** [Azure/azure-sdk-for-java#4868]
**Describe the bug**
Unable to locate the etag in `BlobProperties`.

**To Reproduce**
No runtime reproduction required. This bug is about an (apparently) missing critical feature.

***Code Snippet***
```
BlobClient client = ... //client definition
BlobProperties properties = client.getProperties().value();
// there's no properties.getEtag() like in v8
// there's no properties.headers().etag() like in v11
```

**Expected behavior**
As described in the above code snippet, I am unable to locate the etag and therefore unable to use it to configure blob access conditions.

**Setup (please complete the following information):**
 - OS: Ubuntu 16.04
 - IDE : IntelliJ
 - com.azure:azure-storage-blob:12.0.0-preview.1

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by rickle-msft** [comment]
Hi, @raj2103. Thank you for posting this. I am taking a look now and will follow up shortly.

**Comment by rickle-msft** [comment]
@raj2103. You are correct that the etag property was erroneously left off of BlobProperties. We will fix this oversight in our next preview. In the meantime you should be able to access the headers property on the response (.headers() instead of .value()) and you can get the etag value from there. You'll have to key into a map instead of being able to get a property, but it should hopefully hold you over until next preview.

**Comment by raj2103** [comment]
@rickle-msft sounds good. While preview 3 is not released, what `key_name` can we use to fetch the etag from the `HttpHeaders`?

**Comment by rickle-msft** [comment]
The header comes back from the service as "Etag", so that should work. There's a chance it gets lower cased, so if that doesn't work, you can try "etag". If neither of those works, then let me know.

**Comment by zhusijia26** [comment]
Close due to idle time. Please reopen if there is still question.

**Comment by raj2103** [comment]
@zhusijia26 Not sure what you mean by idle time here. Have etags been included in blob properties? Has the plan to do that been dropped? Would customers always have to key out the etags from http headers?

I would keep this issue open till the fix suggested by @rickle-msft (above) has been merged.

**Comment by zhusijia26** [comment]
@raj2103 we can keep this issue. At the same time, @rickle-msft provided some alternatives to solve the issue. Could you please try if it works? @rickle-msft do you have an ETA for the release?

**Comment by rickle-msft** [comment]
@raj2103 I apologize for not responding earlier. I must have missed the notification for this. I added the appropriate tags so that we should pick it up better in our planning process now. I don't know if this is included in our projected items for preview 3 which has a code freeze next week, but I will double check

**Comment by rickle-msft** [comment]
In any case, this will certainly be in before we GA

**Comment by alzimmermsft** [comment]
@raj2103, this was fixed in PR #5268. If you experience further issues with missing properties please re-open this issue or create another, thank you.
